File: src/gestion/Asignatura.java
package gestion;

import java.util.*;

import personas.Profesor;

public class Asignatura {
	private int creditos;
	private ArrayList<Profesor> profesor;
	private ArrayList<Asignatura> prerrequisitos;
	private String lineaEnfasis;
	private static ArrayList<Asignatura> listaAsignaturas;
	
	
	
	public Asignatura(int creditos, ArrayList<Profesor> profesor, ArrayList<Asignatura> prerrequisitos,
			String lineaEnfasis) {
		super();
		this.creditos = creditos;
		this.profesor = profesor;
		this.prerrequisitos = prerrequisitos;
		this.lineaEnfasis = lineaEnfasis;
	}

	public void displayProfesores() {
		
	}

	public int getCreditos() {
		return creditos;
	}

	public void setCreditos(int creditos) {
		this.creditos = creditos;
	}

	public ArrayList<Profesor> getProfesor() {
		return profesor;
	}

	public void setProfesor(ArrayList<Profesor> profesor) {
		this.profesor = profesor;
	}

	public String getLineaEnfasis() {
		return lineaEnfasis;
	}

	public void setLineaEnfasis(String lineaEnfasis) {
		this.lineaEnfasis = lineaEnfasis;
	}

	public ArrayList<Asignatura> getPrerrequisitos() {
		return prerrequisitos;
	}

	public void setPrerrequisitos(ArrayList<Asignatura> prerrequisitos) {
		this.prerrequisitos = prerrequisitos;
	}

	public static ArrayList<Asignatura> getListaAsignaturas() {
		return listaAsignaturas;
	}

	public static void setListaAsignaturas(ArrayList<Asignatura> listaAsignaturas) {
		Asignatura.listaAsignaturas = listaAsignaturas;
	}


	
}

File: srcpy/gestorAplicacion/gestion/asignatura.py
class Asignatura:
    listaAsiganturas = []

    def __init__(
        self, nombre, creditos, profesor, prerrequisitos, lineaenfasis
    ) -> None:
        self.nombre = nombre
        self.creditos = creditos
        self.profesor = profesor
        self.prerrequisitos = prerrequisitos
        self.lineaEnfasis = lineaenfasis
        Asignatura.listaAsiganturas.append(self)

    def getCreditos(self):
        return self.creditos

    def setCreditos(self, creditos):
        self.creditos = creditos

    def getProfesor(self):
        return self.profesor

    def setProfesor(self, profesor):
        self.profesor = profesor

    def getLineaEnfasis(self):
        return self.lineaEnfasis

    def setLineaEnfasis(self, linea):
        self.lineaEnfasis = linea

    def getPrerequisitos(self):
        return self.prerrequisitos

    def setPrerequisitos(self, prerrequisitos):
        self.prerrequisitos = prerrequisitos
    @classmethod
    def getListaAsignaturas(cls):
        return cls.listaAsiganturas
    @classmethod
    def setListaAsignaturas(cls, lista):
        cls.listaAsiganturas = lista

    def getNombre(self):
        return self.nombre

    def setNombre(self, nombre):
        self.nombre = nombre

    def agregarProfesor(self, profesor):
        self.profesor.append(profesor)
